integration-tests/keycloak-authorization/src/test/java/io/quarkus/it/keycloak/DynamicTenantConfigPolicyEnforcerTest.java

```diff
@@ -0,0 +1,250 @@
+        assureGetPath("//dynamic-permission-tenant", 403, getAccessToken("jdoe"), null);
+
+        assureGetPath("/dynamic-permission-tenant", 200, getAccessToken("admin"), "Permission Resource Tenant");
+        assureGetPath("//dynamic-permission-tenant", 200, getAccessToken("admin"), "Permission Resource Tenant");
+    }
+
+    @Test
+    public void testKeycloakPolicyEnforcerTenantConfigBuilder() {
+        assertBuilderPopulatedWithDefaultValues();
+        assertEveryConfigPropertyCanBeSet();
+        assertTenantConfigEnhanced(enforcerConfig.namedTenants().get("api-permission-tenant"));
+        assertBuilderShortcuts();
+    }
+
+    private static void assertBuilderPopulatedWithDefaultValues() {
+        var config = KeycloakPolicyEnforcerTenantConfig.builder().build();
+        assertEquals(20, config.connectionPoolSize());
+        assertTrue(config.policyEnforcer().lazyLoadPaths());
+        assertEquals(ENFORCING, config.policyEnforcer().enforcementMode());
+        assertTrue(config.policyEnforcer().paths().isEmpty());
+        assertFalse(config.policyEnforcer().httpMethodAsScope());
+        assertEquals(30000, config.policyEnforcer().pathCache().lifespan());
+        assertEquals(1000, config.policyEnforcer().pathCache().maxEntries());
+        assertTrue(config.policyEnforcer().claimInformationPoint().simpleConfig().isEmpty());
+        assertTrue(config.policyEnforcer().claimInformationPoint().complexConfig().isEmpty());
+        // now let's create path and see that all defaults are populated
+        config = KeycloakPolicyEnforcerTenantConfig.builder()
+                .setPaths("/api/one").setMethod("TRACE", null)
+                .setPaths("/api/two").setMethod("TRACE", null)
+                .build();
+        config.policyEnforcer().paths().values().forEach(p -> {
+            assertEquals(ENFORCING, p.enforcementMode());
+            assertTrue(p.claimInformationPoint().complexConfig().isEmpty());
+            assertTrue(p.claimInformationPoint().simpleConfig().isEmpty());
+            assertTrue(p.name().isEmpty());
+            assertEquals(1, p.paths().get().size());
+            assertEquals(1, p.methods().size());
+            var method = p.methods().get("trace");
+            assertNotNull(method);
+            // we didn't add scopes, and we are yet to require them
+            assertTrue(method.scopes().isEmpty());
+            assertEquals("TRACE", method.method());
+            // default scope enforcement is ALL
+            assertEquals(PolicyEnforcerConfig.ScopeEnforcementMode.ALL, method.scopesEnforcementMode());
+        });
+        // create config from previous config and alter one of paths
+        var builder = KeycloakPolicyEnforcerTenantConfig.builder(config);
+        var path = builder.setPaths("/api/one");
+        path.setPermissionName("New Permission");
+        path.setClaimInformationPoint(Map.of("key", Map.of("sub-key", "sub-value")));
+        path.clearMethods();
```

Comment:
I'm not sure it is necessary. Users will set up the config and build it, in a client scope, where they can reuse the same builder instance for creating multiple configurations, having clear methods might be useful, but not in the server scope IMHO. `SecurityIdentity` builder does not have clear methods for example

Reply:
I added it for very edge case, in case you just want to adjust original configuration (remove same methods without rewriting them). I'll drop it. It's no biggie.

Reply:
Done.

Reply:
Removed.